android/src/main/java/com/swmansion/gesturehandler/core/GestureUtils.kt

```diff
@@ -47,4 +55,101 @@ object GestureUtils {
   }
   fun coneToDeviation(angle: Double): Double =
     cos(Math.toRadians(angle / 2.0))
+
+  fun updateStylusData(stylusData: StylusData, event: MotionEvent) {
+    // On web and iOS 0 degrees means that stylus is parallel to the surface. On android this value will be PI / 2.
+    val altitudeAngle = (PI / 2) - event.getAxisValue(MotionEvent.AXIS_TILT).toDouble()
+    val pressure = event.getPressure(0).toDouble()
+
+    var orientation = event.getOrientation(0).toDouble()
+
+    // To get azimuth angle, we need to use orientation property. Orientation value range is [-PI, PI] (https://developer.android.com/develop/ui/compose/touch-input/stylus-input/advanced-stylus-features#orientation)
+    // To shift range into [0, 2PI], we add 2PI if orientation is less than 0.
+    if (orientation < 0) {
+      orientation += 2 * PI
+    }
+
+    // To get the same value that we get on web, we have to perform shift by PI/2.
+    // However, if orientation is greater than 3PI/2 we would get angles greater than 2PI. Therefore we simply subtract 3PI/2 in that case.
+    val azimuthAngle = if (orientation >= 3 * PI / 2) {
+      orientation - 3 * PI / 2
+    } else {
+      orientation + PI / 2
+    }
+
+    val tilts = spherical2tilt(altitudeAngle, azimuthAngle)
+
+    stylusData.tiltX = tilts.first
+    stylusData.tiltY = tilts.second
+    stylusData.altitudeAngle = altitudeAngle
+    stylusData.azimuthAngle = azimuthAngle
+    stylusData.pressure = pressure
```

Comment:
I think it would be better to assign a new object here. Since it's passed referentially to the `EventDataBuilder,` changes like that may modify an already scheduled but not yet sent event. (This would also warrant a name change)